pkg/common.go

```diff
@@ -84,24 +87,32 @@ func GetNamespace(csi *csibaremetalv1.Deployment) string {
 	return csi.Namespace
 }
 
-func isDaemonSetDeployed(dsClient appsv1.DaemonSetInterface, name string) (bool, error) {
-	_, err := dsClient.Get(name, metav1.GetOptions{})
-	return isFound(err)
-}
+func deploymentChanged(expected *v1.Deployment, found *v1.Deployment) bool {
+	if !equality.Semantic.DeepEqual(expected.Spec.Replicas, found.Spec.Replicas) {
```

Comment:
How about version and other checks?

Reply:
@mishoyama version and all other parameters are set in instance.Spec.Template (includes PodSpec) and I check it in `deploymentChanged`

Reply:
Ok